tests/test_mathematics.py

```diff
@@ -0,0 +1,93 @@
+import warnings
+import cupy as cp
+import pytest
+
+with warnings.catch_warnings():
+    warnings.filterwarnings(
+        action="ignore", category=UserWarning, message=r"matplotlib not found:"
+    )
+    import qutip.tests.core.data.test_mathematics as test_tools
+
+from qutip_cupy import dense  # noqa:E402
+from qutip_cupy import dense_functions as cdf  # noqa:E402
+from qutip_cupy import CuPyDense  # noqa:E402
```

Comment:
Same comment about moving the qutip import after importing qutip_cupy and removing the `# noqa`s and warning filter.

Reply:
After checking we do not need this filter in particular, and there is no noqa comment needed then. 
But @hodgestar , from past comments , is the correct order base python libraries, external libraries, our own modules?

Reply:
By default, yes -- Python standard library, then third party packages, then this package. However, sometimes exceptions can be made if needed in strange cases such as this, which is why I suggested keeping a comment.

Note that if you run all the tests with `pytest tests` then only the first test file would need the strange ordering but I recommend keeping the strange ordering in all test files so that they can also be run individually with `pytest tests/test_foo.py`.